### DISCUSSION TITLE
Add i18n support for site title

### DISCUSSION THREAD

**Discussion by ybw0014** [withastro/starlight#1282] [Feature Requests]
### What version of `starlight` are you using?

0.15.0

### What is your idea?

Allows site title to be translated by i18n.

### Why is this feature necessary?

Current i18n is fine, but I would like the site title to be able to be translated as well.

### Do you have examples of this feature in other projects?

My site's title is `ybw0014's Minecraft plugin documents`. It should be translated in other languages. It is not like a product name that can choose not to translate.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for the feature request @ybw0014! I agree we should support this.

For some context, we did previously start work on support for this in #281 but agreed we’d prefer an approach where fields like this can either be a simple string as currently:

```js
title: 'My site'
```

OR an object defining the translations:

```js
title: {
  en: 'My site',
  fr: 'Mon site',
},
```

i.e. avoid introducing dedicated fields for translations (e.g. `title` and `titleTranslations`)

So that would probably be the API we’d like to implement in case someone is interested in contributing a PR.

**Comment by HiDeoo** [comment]
Starlight [`0.22.0`](https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.22.0) added support for translating the site title.

Documentation for this feature is available [here](https://starlight.astro.build/guides/i18n/#translate-the-site-title) but the change would be:

```diff
import { defineConfig } from 'astro/config';
import starlight from '@astrojs/starlight';

export default defineConfig({
  integrations: [
    starlight({
-      title: 'My Docs',
+      title: {
+         en: 'My Docs',
+        'zh-CN': '我的文档',
+     },
      defaultLocale: 'en',
      locales: {
        en: { label: 'English' },
        'zh-cn': { label: '简体中文', lang: 'zh-CN' },
      },
    }),
  ],
});
```
